lib/SPIRV/SPIRVReader.cpp

```diff
@@ -694,6 +695,108 @@ void SPIRVToLLVM::setLLVMLoopMetadata(LoopInstType *LM, BranchInst *BI) {
   BI->setMetadata("llvm.loop", Node);
 }
 
+// Checks whether target basic block is reachable from current basic block.
+// CB - current basic block.
+// TBId - Id of target basic block.
+// VBBIds - set of Ids of already visited basic blocks.
+static bool isBBReachableFrom(const SPIRVValue *CB, SPIRVId TBId,
+                              std::unordered_set<SPIRVId> &VBBIds) {
+  if (!CB || !CB->isBasicBlock())
+    return false;
+
+  if (CB->getId() == TBId)
+    return true;
+
+  auto InsRes = VBBIds.insert(CB->getId());
+  if (!InsRes.second)
+    return false;
+
+  const auto *CBB = static_cast<const SPIRVBasicBlock *>(CB);
+  const auto *TI = CBB->getTerminateInstr();
+  if (!TI)
+    return false;
+
+  switch (TI->getOpCode()) {
+  case OpBranch: {
+    const auto *BR = static_cast<const SPIRVBranch *>(TI);
+    return isBBReachableFrom(BR->getTargetLabel(), TBId, VBBIds);
+  }
+
+  case OpBranchConditional: {
+    const auto *BR = static_cast<const SPIRVBranchConditional *>(TI);
+    return (isBBReachableFrom(BR->getTrueLabel(), TBId, VBBIds) ||
+            isBBReachableFrom(BR->getFalseLabel(), TBId, VBBIds));
+  }
+
+  case OpSwitch: {
+    const auto *BS = static_cast<const SPIRVSwitch *>(TI);
+    for (const auto *BV : BS->getPairs()) {
+      if (isBBReachableFrom(BV, TBId, VBBIds))
+        return true;
+    }
+    return false;
+  }
+
+  default:
+    return false;
+  }
+}
+
+static inline bool isBBReachableFrom(const SPIRVValue *CB, SPIRVId TBId) {
+  std::unordered_set<SPIRVId> VBBIds{};
```

Comment:
`std::unordered_set<SPIRVId> VBBIds;` is enough

Reply:
Thank you. Fixed.